Add Amazon Bedrock support

kmotohas opened the pull request:
# Overview
This pull request adds support for Amazon Bedrock, AWS's generative AI service, as a backend for LLM and embedding models in nano-graphrag.

# Changes
- Integration of Asynchronous AWS SDK for Python (aioboto3)
- Implementation of LLM interface for Amazon Bedrock
  - Added support for Claude 3.5 Haiku, Claude 3 Haiku, and Claude 3 Sonnet models
  - Implemented `amazon_bedrock_complete_if_cache` function for model inference
- Implementation of embedding interface for Amazon Bedrock
  - Added amazon_bedrock_embedding function using Titan Embed Text v2 model
- Updated configuration options to include Amazon Bedrock support
- Modified existing functions to accommodate Amazon Bedrock integration

# How to Test
1. Set up AWS credentials:

```bash:
export AWS_ACCESS_KEY_ID=your_access_key
export AWS_SECRET_ACCESS_KEY=your_secret_key
export AWS_REGION=us-east-1  # or your preferred region
```

2. Update your code to use Amazon Bedrock models and embeddings

```bash:
graph_func = GraphRAG(working_dir="./working_dir", using_amazon_bedrock=True)
```

# Note

- Ensure proper IAM permissions are set up for accessing Amazon Bedrock

gusye1234 commented:
LGTM! Would you mind update the readme as well? Maybe add a tip at https://github.com/gusye1234/nano-graphrag?tab=readme-ov-file#quick-start, So people know they can use Claude through Amazon Bedrock

kmotohas commented:
@gusye1234 Thank you for the review! I have added tips to use Amazon Bedrock in README.md. I also modified to use a factory function to create functions for Amazon Bedrock completion instead of defining one for each model ID to keep pace with service updates.
